File: zoho/L165-CompareVersionNumber.java
// https://leetcode.com/problems/compare-version-numbers/

class Solution {
    public int compareVersion(String version1, String version2) {
        int p1 = 0, p2 = 0;
        int num1 = 0, num2 = 0;
        int len1 = version1.length(), len2 = version2.length();
        while (p1 < len1 || p2 < len2) {
            num1 = 0;
            num2 = 0;
            while (p1 < len1 && version1.charAt(p1) != '.') {
                num1 = (num1 * 10 + (version1.charAt(p1) -'0'));
                p1++;
            }
            while (p2 <len2 && version2.charAt(p2) != '.') {
                num2 = (num2 * 10 + (version2.charAt(p2) - '0'));
                p2++;
            }
            if (num1 < num2)
                return -1;
            else if (num1 > num2)
                return 1;
            else {
                p1++;
                p2++;
            }
        }
        return 0;
    }
}

File: zoho/L165-compare_version_number.py
# https://leetcode.com/problems/compare-version-numbers/
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        rev1, rev2 = 0, 0
        p1, p2 = 0, 0
        numVal = ord('0')
        len1, len2 = len(version1), len(version2)
        while (p1 < len1 or p2 < len2):
            rev1, rev2 = 0 , 0
            while (p1 < len1 and version1[p1] != '.'):
                rev1 = (rev1 * 10 + (ord(version1[p1]) - numVal))
                p1 += 1
            while (p2 < len2 and version2[p2] != '.'):
                rev2 = (rev2 * 10 + (ord(version2[p2]) - numVal))
                p2 += 1
            if (rev1 > rev2):
                return 1
            elif (rev2 > rev1):
                return -1
            else:
                p1 += 1
                p2 += 1
        return 0
